src/libraries/System.Runtime/tests/System/ActivatorTests.Generic.cs

```diff
@@ -60,6 +76,82 @@ public partial class ActivatorTests
         public void CreateInstanceT_StructWithDefaultConstructorThatThrows_ThrowsTargetInvocationException() =>
             Assert.Throws<TargetInvocationException>(() => Activator.CreateInstance<StructWithDefaultConstructorThatThrows>());
 
+        [Fact]
+        public void CreateFactoryT_ReferenceTypeWithPublicCtor_Success()
+        {
+            Func<PublicType> factory = Activator.CreateFactory<PublicType>();
+            PublicType instance = factory();
+            Assert.NotNull(instance);
+            Assert.True(instance.CtorWasCalled);
+        }
+
+        [Fact]
+        public void CreateFactoryT_ReferenceTypeWithPrivateCtor_ThrowsMissingMethodException()
+            => Assert.Throws<MissingMethodException>(() => Activator.CreateFactory<PrivateTypeWithDefaultCtor>());
+
+        [Fact]
+        public void CreateFactoryT_ReferenceTypeWithPrivateCtorThatThrows_DoesNotWrapInTargetInvocationException()
+        {
+            Func<TypeWithDefaultCtorThatThrows> factory = Activator.CreateFactory<TypeWithDefaultCtorThatThrows>();
+            Assert.Throws<Exception>(() => factory()); // no TIE
+        }
+
+        [Fact]
+        public void CreateFactoryT_OfNullableT_ReturnsNull()
+        {
+            Func<int?> factory = Activator.CreateFactory<int?>();
+            Assert.NotNull(factory);
+            int? instance = factory();
+            Assert.False(instance.HasValue);
+        }
+
+        [Fact]
+        public void CreateFactoryT_OfValueTypeWithoutDefaultCtor_ReturnsDefaultT()
+        {
+            Func<ValueTypeWithParameterfulConstructor> factory = Activator.CreateFactory<ValueTypeWithParameterfulConstructor>();
+            Assert.NotNull(factory);
+            ValueTypeWithParameterfulConstructor instance = factory();
+            Assert.False(instance.ParameterfulCtorWasCalled);
+        }
+
+        [Fact]
+        public void CreateFactoryT_OfValueTypeWithPublicDefaultCtor_CallsDefaultCtor()
+        {
+            Func<StructWithPublicDefaultConstructor> factory = Activator.CreateFactory<StructWithPublicDefaultConstructor>();
+            Assert.NotNull(factory);
+            StructWithPublicDefaultConstructor instance = factory();
+            Assert.True(instance.ConstructorInvoked);
+            Assert.True(IsAddressOnLocalStack(instance.AddressPassedToConstructor)); // using Func<T>, value type ctor is called using ref to stack local, no boxing
```

Comment:
This is implementation detail. I can imagine some imagine some implementations may choose to allocate the box to keep things simple.